TEST_APPS/testcases/netsocket/TCPSOCKET_ACCEPT.py

```diff
@@ -0,0 +1,102 @@
+                       type="smoke",
+                       subtype="socket",
+                       requirements={
+                           "duts": {
+                               '*': { #requirements for all nodes
+                                    "count":2,
+                                    "type": "hardware"
+                                },
+                                "1": {"nick": "dut1"},
+                                "2": {"nick": "dut2"}
+                           }
+                       }
+        )
+
+    def setup(self):
+        interface = interfaceUp(self, ["dut1"])
+        self.server_ip = interface["dut1"]["ip"]
+        interface = interfaceUp(self, ["dut2"])
+        self.client_ip = interface["dut2"]["ip"]
+
+    def clientThread(self):
+        self.logger.info("Starting")
+        time.sleep(5) #wait accept from server
+        self.command("dut2", "socket " + str(self.client_socket_id) + " open")
+        self.command("dut2", "socket " + str(self.client_socket_id) + " connect " + str(self.server_ip) + " " + str(self.used_port))
+
+
+    def case(self):
+        self.used_port = 2000
+
+        response = self.command("dut1", "socket new TCPSocket")
+        self.server_base_socket_id = int(response.parsed['socket_id'])
+
+        self.command("dut1", "socket " + str(self.server_base_socket_id) + " open")
+        response = self.command("dut1", "socket " + str(self.server_base_socket_id) + " bind port " + str(self.used_port), report_cmd_fail = False)
+        if response.retcode == -1:
+            if (response.verify_trace("NSAPI_ERROR_UNSUPPORTED", break_in_fail = False)):
+                raise SkippedTestcaseException("UNSUPPORTED")
+            else:
+                TestStepFail("Bind port failed")
+        self.command("dut1", "socket " + str(self.server_base_socket_id) + " listen")
+
+        response = self.command("dut2", "socket new TCPSocket")
+        self.client_socket_id = int(response.parsed['socket_id'])
+
+        #Create a thread which calls client connect()
+        t = threading.Thread(name='clientThread', target=self.clientThread)
+        t.start()
+
+        response = self.command("dut1", "socket " + str(self.server_base_socket_id) + " accept")
+
+        t.join()
+        socket_id = int(response.parsed['socket_id'])
+        if response.timedelta < 5.0: # Check that socket accept call blocks
+            raise TestStepFail("Longer response time expected")
+
+        self.command("dut1", "socket " + str(socket_id) + " send hello")
+
+        response = self.command("dut2", "socket " + str(self.client_socket_id) + " recv 5")
+        data = response.parsed['data'].replace(":","")
+
+        if data != "hello":
+            raise TestStepFail("Received data doesn't match the sent data")
+
+        response = self.command("dut1", "socket " + str(self.server_base_socket_id) + " close")
+        response = self.command("dut1", "socket " + str(socket_id) + " close")
```

Comment:
Could you please move these close-commands to teardown section.
If the test case raise error, then the close-commands are never called